Assets/Scripts/MusicManager.cs

```diff
@@ -0,0 +1,222 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using PixelCrushers.DialogueSystem;
+using System;
+
+/// <summary>
+/// Manages the background music at runtime
+/// </summary>
+public class MusicManager : MonoBehaviour
+{
+    /// <summary>
+    /// The current state of the town, based on quests completed
+    /// Determines which of the four tracks to play
+    /// </summary>
+    private enum TownMusicState { Cursed, Dreary, Light, Cheery };
+
+    /// <summary>
+    /// Which theme should currently be playing?
+    /// </summary>
+    public enum CurrentMusic { Town, Cave, Forest, Castle };
+
+    [SerializeField]
+    [Tooltip("Music to play while in Gratia proper.")]
+    private List<AudioClip> townMusic;
+
+    [SerializeField]
+    [Tooltip("Music to play in the cave.")]
+    private AudioClip caveMusic;
+
+    [SerializeField]
+    [Tooltip("Music to play in the forest.")]
+    private AudioClip forestMusic;
+
+    [SerializeField]
+    [Tooltip("Music to play in the castle.")]
+    private AudioClip castleMusic;
+
+    [SerializeField]
+    [Tooltip("Time to fade music in/out for track transition.")]
+    private float musicFadeSeconds = 0.25f;
+
+    [SerializeField]
+    [Tooltip("Margin for fading.")]
+    private float fadeMargin = 0.1f;
+
+    [SerializeField]
+    [Tooltip("Completed quest count variable (Dialogue System).")]
+    private string completedQuestsName;
+
+    /// <summary>
+    /// How many quests has the player completed?
+    /// Sets current town music state based on the answer
+    /// </summary>
+    private int QuestsCompleted
+    {
+        get { return questsCompleted_UseProperty; }
+        set
+        {
+            questsCompleted_UseProperty = value;
+            switch(currentMusicState)
+            {
+                case TownMusicState.Cursed:
+                    if (QuestsCompleted >= 2)
+                        isMusicFading = true;
+                    break;
+                case TownMusicState.Dreary:
+                    if(QuestsCompleted >= 5)
+                        isMusicFading = true;
+                    break;
+                case TownMusicState.Light:
+                    if(QuestsCompleted >= 12)
+                        isMusicFading = true;
+                    break;
+            }
+            if(isMusicFading)
+            {
+                isFadingOut = true;
+                currentMusicState++;
+                fadeStartTime = Time.time;
+            }
+        }
+    }
+
+    private CurrentMusic CurrentlyPlayingMusic
+    {
+        get { return currentlyPlayingMusic_UseProperty; }
+        set
+        {
+            if(value != currentlyPlayingMusic_UseProperty)
+            {
+                isMusicFading = true;
+                isFadingOut = true;
```

Comment:
I feel like this logic is somewhat duplicated in the other property. Make it a function, call the function twice.